injector/src/main/java/edu/ucr/cs/riple/injector/Helper.java

```diff
@@ -161,26 +164,31 @@ public static Node findTopLevelDirectInnerClass(Node cursor, String name) {
             nodes.add(node);
           }
         });
-    Preconditions.checkArgument(
-        nodes.size() > 0, "Could not find inner class " + name + " on:\n" + cursor);
+    if (nodes.size() == 0) {
+      throw new TargetClassNotFound("Top-Level-Direct-Inner-Class", name, cursor);
+    }
     return nodes.get(0);
   }
 
-  private static Node findTopLevelInnerClass(Node cursor, String name, int index) {
+  private static Node findTopLevelInnerClass(Node cursor, String name, int index)
+      throws TargetClassNotFound {
     final List<Node> candidates = new ArrayList<>();
     walk(
         cursor,
         candidates,
         node -> {
           Optional<Node> parent = node.getParentNode();
-          if (!parent.isPresent()) {
+          if (parent.isEmpty()) {
             return false;
           }
           if (parent.get().equals(cursor)) {
```

Comment:
Question: What is this line checking for?

Reply:
Actually this was needed for previous approach, with the new `walk` method I wrote, this is not required anymore. Deleted it. 92b769bb4cd4e901a774a89f74e3d690f9cc40d6